src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -134,39 +134,100 @@ static auto getSetUnion(const std::vector<BlockMetadata>& blocks1,
 //______________________________________________________________________________
 // Refers to an `ValueId` vector `::iterator` (index).
 using RandomIt = std::vector<ValueId>::iterator;
+
 // A pair containing the `start` and `end` (iterator) indices for a specified
 // single datatype range over a given `ValueId` vector.
 // For the given cases below, the `ValueId` vector is retrieved from the
 // `BlockMetadata` values. And those `ValueId` values define the bounds of the
 // related `BlockMetadata` values.
 using RandomItPair = std::pair<RandomIt, RandomIt>;
 
+// Refers to a `BlockMetadata` span `::iterator` (index).
+using BlockSpanIt =
+    std::span<const prefilterExpressions::BlockMetadata>::iterator;
+
+//______________________________________________________________________________
+static std::pair<BlockSpanIt, BlockSpanIt> getInputAdjustedRangeIndices(
```

Comment:
Best document this with examples.

Reply:
Or we make a more explicit interface that humans can understand, 
by transforming back the flat `vector<ValueId>` to something understandable like
[block 3, partial], [block 4-123, completely], [block 124, partial] 

With this representation, both the positive and the inverted case become trivial and readable, and you have the 
integer magic completely capsuled, and can work with indices instead of iterators etc.